avionics/common/include/rocket.h

```diff
@@ -0,0 +1,124 @@
+/**
+ * Overarching object that carries different components.
+ */
+#pragma once
+
+#include "CSVWriteImpl.h"
+#include "CSVwrite.h"
+#include "buzzer.h"
+#include "calculations.h"
+#include "cameras.h"
+#include "radio.h"
+#include "sensor_collection.h"
+#include "state_machine.h"
+
+struct Rocket {
+  public:
+    static constexpr char LOG_FILE_NAME[] = "datalog.csv";
+
+    Rocket()
+        : cam(Hal::SerialInst::Camera), datalog(LOG_FILE_NAME),
+          init_status(collectStatus(sensors, ignitors)),
+          radio(Hal::SerialInst::Radio), calc(sensors) {}
+
+    // WARNING - MEMBER ORDER DEPENDENCY
+    // https://isocpp.org/wiki/faq/ctors#order-dependency-in-members
+    Buzzer buzzer;
+    Camera cam;
+    SensorCollection sensors;
+    IgnitorCollection ignitors;
+    CSVWrite<CSVWriteImpl> datalog;
+    RocketStatus init_status;
+    RadioController radio;
+    Calculator calc;
+    StateMachine state_machine;
+
+    // Rocket also acts as radio command receiver
```

Comment:
Hmm, I'm uncertain if this should be the rocket object's job or someone else's.

Reply:
thing is - whatever acts as the command receiver needs to have access to everything. I guess alternative would be e.g. use a nested or friend class with only the `operator()` method.

Reply:
In OOP terms that just means it needs to be one level above the rocket object. I.e. it has a pointer to the rocket object. 

In any case, I think we should avoid operator overloading. It doesn't have any benefits over a simple method and just makes things more confusing to a new reader.